### DISCUSSION TITLE
How to render code-first AppSync GraphQL schema definition to schema string?

### DISCUSSION THREAD

**Discussion by moltar** [aws/aws-cdk#14966] [Q&A] [answered]
I'm using the [code-first](https://docs.aws.amazon.com/cdk/api/latest/docs/aws-appsync-readme.html#code-first-schema) approach for defining the GraphQL schema.

This helps me to reuse existing constants and types when defining the schema.

However, I also want to run [GraphQL Codegen](https://graphql-code-generator.com/) to produce types for other uses cases, such as Lambda specifics and for the frontend clients.

What I am missing is a way to render the code-first schema back to a `.graphql` schema file (or in memory).

Is that possible?

Ping: @BryanPan342 any ideas?

Thanks!

**Comment by BryanPan342** [comment, marked as answer]
Yup! I think this would be a good feature to add and something we mentioned in [this issue](https://github.com/aws/aws-cdk/issues/14022#issuecomment-834777782)

But I found a way to eject it manually within your own CDK app:

https://github.com/BryanPan342/starwars-code-first/blob/9efb83798f9127423edc0f02d6a21fd87b8dd0ab/bin/starwars-code-first.ts#L10-L19

**Reply by moltar** [reply]
Thanks for the eject idea! Will try :)
